### PR TITLE
Add support for SSL

### PR DESCRIPTION
**Pull request by erikseppanen**
Add support for java mongo driver parameters for SSL.

### CODE REVIEW

File: test/monger/test/core_test.clj [hunk 1]
```diff
@@ -66,6 +66,8 @@
               :auto-connect-retry true
               :max-auto-connect-retry-time 1
               :description "Description"
+              :ssl-enabled true
+              :ssl-invalid-hostname-allowed true
```
Comment (michaelklishin) [inline test/monger/test/core_test.clj]: Seriously?

File: test/monger/test/core_test.clj [hunk 2]
```diff
@@ -66,6 +66,8 @@
               :auto-connect-retry true
               :max-auto-connect-retry-time 1
               :description "Description"
+              :ssl-enabled true
```
Comment (michaelklishin) [inline test/monger/test/core_test.clj]: Yeah, lets not use this in existing tests.

**Comment by michaelklishin** [comment]
This is "the most simplistic TLS support" at best. How will client and CA certificates/keys be provided?

**Comment by michaelklishin** [comment]
I'm not going to merge this. While I appreciate you taking the time, this is not TLS support, this is a joke.

TLS support involves providing certificates and private keys, which on the JVM can be fairly finicky (depends on the library). It also needs a separate set of tests with locally generated certificates, not "just trust any peer" on for every test.

**Comment by bm3719** [comment]
I'm also interested in this feature if we could get it in, either in the form of this pull request or some other way.

**Comment by gilburd** [comment]
Maybe add an option to set sslContext? Much simpler than using jvm's trust store. 
That's what I'm using to connect to AWS's documentDB. 
Should I send a pull request? 
https://github.com/gilburd/monger/commit/cd730a0ec784a9cb6dd1f4410bc6da71375cfefe

**Comment by erikseppanen** [comment]
fwiw, the code for this pull request was added years later:
- https://github.com/michaelklishin/monger/commit/ccffffc9125dcdcf305f98a1259a1b7dbec29e73
- https://github.com/michaelklishin/monger/commit/010a977aacd964b2967189634e5c932548aa21b1

**Pull request closed without merging**